### ISSUE TITLE
OfficeRuntime.auth.getAccessToken hangs in Outlook Event-Based Activation

### ISSUE THREAD

**Issue by barclayadam** [OfficeDev/office-js#4104]
# Provide required information needed to triage your issue

When using the `OfficeRuntime.auth.getAccessToken` API in event-based activation we cannot fall back to other mechanisms because it completely hangs the script, never returning control.

## Your Environment

* Platform: PC Desktop
* Host: Outlook
* Office version number: 16.0.17318.20000
* Operating System: Windows
 
## Expected behavior

The Promise from `OfficeRuntime.auth.getAccessToken` is rejected with an error that our code can handle and fall back to alternative mechanisms.

## Current behavior

An error is logged to the runtime log, but control is never returned to the add-in. This is only an issue in Outlook PC Event-Based activation. The task pane and OWA correctly respond with an error.

In PC event based activation we see:

````
06/02/2024 10:45:51	Verbose	Runtime	[Console] [Log] [S365] [messageCompose] Getting access token		
06/02/2024 10:45:51	Verbose	Runtime	[Console] [Log] MailboxHostExecuteApi invoked!		
06/02/2024 10:45:51	Monitorable	WebApplicationInfo	SSO failed with "5001" error code.		
06/02/2024 10:45:56	Monitorable	WebApplicationInfo	SSO failed with "13007" error code.	
````

I do not always see the 5001 error, I have only seen that on a single run. You can see that we log a message, then Outlook itself, but nothing further is logged.

In other contexts, we get the error: `Error occurred in the authentication provider.: An unexpected error occurred on the server. (13007)`

Note that I believe the actual reason for failing is because of mismatched URIs between local add-in and Entra ID registration, but the cause is not the issue here, it's the lack of response so I have not attempted to fix the issue. I have seen this happen on another machine too which I do not believe will be the same reason.

## Steps to reproduce

In event-based activation run the below code:

````typescript
try {
    return OfficeRuntime.auth.getAccessToken()
        .then((token) => {
            console.info('Exchanging');

            return exchangeAadSsoToken(cache, token);
        })
        .catch(err => {
            console.info('Fallback from Promise');

            return exchangeUserIdentityFallback(err, cache);
        })
        .then(r => r.value);
} catch (err) {
    console.info('Fallback from catch');

    return exchangeUserIdentityFallback(err, cache).then(r => r.value);
}
````

I've not included details of the called methods, they are irrelevant to this issue as execution never gets that far.

## Link to live example(s)

I do not have a live example, but can capture any time travel traces or similar as necessary to give you more context for the failure.

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by glr0221** [comment]
@exextoc 

I am seeing this as well with our event-based-addin implementation. 3 weeks ago, this was working fine and I was able to use GRAPH-SSO in my event-based-addin. But today, when I tested the same code, it got stuck with the 'OfficeRuntime.auth.getAccessToken' call. This only happens in Windows Outlook clients.

I hope you can take a look into this. 

Thanks.

**Comment by mqi-diamonddiagnostics-com** [comment]
If your add-in file is hosted under a path (excluding domain) containing dot(.), please change it to hyphen(-).
Remember to update every resource and most importantly .well-known.
This workaround worked for me, even though I am still suffering from a (block:other) issue on web client. At least Windows client worked.

**Comment by glr0221** [comment]
Thanks @mqi-diamonddiagnostics-com. 

Currently my addin is hosted in the following url : 

`https://aa-bb-cccc.dd.ee.ff.com/gg/wwwwww/xxxxxxxxx.js`

I think it fits the format you said. Or is my url is not good enough? 
thanks in advance.

**Comment by barclayadam** [comment]
That is not the issue in our case. The file is "https://outlook.signature365.com/commandsdesktop.js". `.well-known` endpoint points to this file too.

Regardless of the error, the call to `OfficeRuntime.auth.getAccessToken` should **never** hang completely

**Comment by mqi-diamonddiagnostics-com** [comment]
> Thanks @mqi-diamonddiagnostics-com.
> 
> Currently my addin is hosted in the following url :
> 
> `https://aa-bb-cccc.dd.ee.ff.com/gg/wwwwww/xxxxxxxxx.js`
> 
> I think it fits the format you said. Or is my url is not good enough? thanks in advance.

It looks good to me. I used the pattern `https://www.abcdefgh.com/<app-id>/<major-minor-patch-build>/launchevent.js`. The host domain is SSL certified via AWS.
I also figured out yesterday that certain `<app-id>` (folked from their original react copy without regeneration) will be blocked by Chrome or its NewOutlook on Windows. I updated my app-id with a random new one and updated my host paths. Now my add-in functions okay in sideloading test.

**Comment by glr0221** [comment]
@exextoc 

Has your team already reproduced the issue? This is currently blocking attempts to implement GRAPH in event-based addins. 

Thank you so much in advance.

**Comment by ztcdsb** [comment]
@glr0221 @barclayadam A few weeks ago, there was an authentication-related issue. Could you please test it again and let us know if the problem persists? 
Since we haven't been able to consistently reproduce this issue, it would be ideal if you could share a live example with us.

**Comment by glr0221** [comment]
@ztcdsb thank you for the update. I am currently working on spam addins and OfficeRuntime.auth.getAccessToken is not working correctly. It used to work 3-4 weeks ago. 

I have filed a new ticket : [https://github.com/OfficeDev/office-js/issues/4211](https://github.com/OfficeDev/office-js/issues/4211)

**Comment by PP-Admin** [comment]
We are experiencing this issue as well with an Outlook add-in, and it started around the same time as mentioned above... we have found that the issue appears when customers are on version 2402. If you roll back to 2401 it works again. So it looks like the issue is in 2402 (if it's the same issue, there could be multiple things going on, we're still investigating).  But a fix at the Microsoft end would be great!

**Comment by glr0221** [comment]
thanks for sharing this @PP-Admin . I will try this on my setup as well. Thank you again.

**Comment by glr0221** [comment]
@PP-Admin I have been trying different subversions of 2401 but to no success. Would you be kind enough to share the exact subversion of 2401 which worked for you? Thank you so much.

**Comment by PP-Admin** [comment]
Hi @glr0221 we used the following command to roll back and that worked.  
Unfortunately later releases also contain the bug, we have customers reporting the issue on 2403 as well now. Rolling back the version eliminates the problem, for the purpose of testing, but it's no solution for us, as we can't make customers roll back. 

"C:\Program Files\Common Files\microsoft shared\ClickToRun\officec2rclient.exe" /update user updatetoversion=16.0.17231.20236

**Comment by SimonDania** [comment]
We are experiencing a similar issue with our add-in.
When we are calling `OfficeRuntime.auth.getAccessToken` through event-based activation, we are just getting an error 5001.
The issue is present on version 2402 of Outlook, and rolling back to 2401 solves the issue, so it has to be an error that's been introduced with 2402.
We're currently investigating if we can do anything in our end, but would like to hear from @exextoc if this is something they are looking into or not.

**Comment by barclayadam** [comment]
I have tested our add-in locally just now, this bug still exists and still hangs, not returning the execution back to our add-in. This **does** work in the task pane of the same build (slightly newer than the original now at `16.0.17512.20000`).

The runtime log, as above just prints a failure message and gives up:

````
20/03/2024 14:17:06	Verbose	Runtime	[Console] [Log] MailboxHostExecuteApi invoked!		
20/03/2024 14:17:06	Monitorable	WebApplicationInfo	SSO failed with "13007" error code.		
20/03/2024 14:18:36	Monitorable	WebApplicationInfo	SSO failed with "13007" error code.		
20/03/2024 14:18:37	Monitorable	WebApplicationInfo	SSO failed with "13007" error code.		
20/03/2024 14:18:41	Monitorable	WebApplicationInfo	SSO failed with "13007" error code.		
20/03/2024 14:18:49	Monitorable	WebApplicationInfo	SSO failed with "13007" error code.		
20/03/2024 14:18:50	Monitorable	WebApplicationInfo	SSO failed with "13007" error code.	
````

Note that the above represents a single run and failure, so perhaps the issue is one of a retry mechanism that is not returning control to the client?

I'm pretty sure the actual root cause of this issue is a configuration error that can be resolved, but I need the confidence that if there are errors for any reason our add-in would have control given back to allow a fallback, otherwise I cannot with confidence take this to production

**Comment by ztcdsb** [comment]
The issue has been resolved. Please let me know if you still experience it.

**Comment by glr0221** [comment]
@ztcdsb 

Which build was this resolved in? I am currently using Beta version : Version 2404 (Build 17531.20000). 

I am still experiencing the hanging issue with OfficeRuntime.auth.getAccessToken(). This is when I use it for the spam addin which is basically an event-based addin. 

Thank you.

**Comment by ztcdsb** [comment]
@glr0221 Apologies for the confusion—there have been a few token-related issues, and I mixed them up. 
The underlying problem is a configuration error, and we're currently working on resolving it. I'll provide further updates once the fix is implemented.

**Comment by glr0221** [comment]
any update on this item @ztcdsb ? Thanks

**Comment by MrDaleQ** [comment]
@glr0221 

FYI we've been informed that this will be resolved by the end of the month through other channels.

**Comment by glr0221** [comment]
thanks @MrDaleQ

**Comment by glr0221** [comment]
Any update on this? I have tried the latest version of the current channel and still, invoking OfficeRuntime.auth.getAccessToken() gets stuck resulting to my spam addin loop in processing and eventually time out. Anybody got OfficeRuntime.auth.getAccessToken() working for their spam addins?

Thanks

**Comment by ronnievdv** [comment]
Any update on this?  I get the same error logged, but the promise catch / then is never called when used in an event, but they are called when using it in a task pane. 
Outlook Version 2501 Build 18429.20132.
```javascript
  try {
    OfficeRuntime.auth
      .getAccessToken({
        allowSignInPrompt: true,
        allowConsentPrompt: true,
        forMSGraphAccess: true,
      })
      .then((response) => {
        console.log(response); // this is never called
      })
      .catch((err) => {
        console.log(err); // this is never called
      });
  } catch (exception) {
     console.log(err); // this is never called
  }
```

**Comment by kwame-ansong** [comment]
We have internal tracking ID 9721273 for this item.
Unfortunately, we cannot share any timelines here.
